src/analysis/normed_space/normed_group_hom_completion.lean

```diff
@@ -0,0 +1,259 @@
+variables {H : Type*} [semi_normed_group H]
+variables {K : Type*} [semi_normed_group K]
+
+/-- The normed group hom induced between completions. -/
+def normed_group_hom.completion (f : normed_group_hom G H) :
+  normed_group_hom (completion G) (completion H) :=
+{ bound' := begin
+    use ∥f∥,
+    intro y,
+    apply completion.induction_on y,
+    { exact is_closed_le
+            (continuous_norm.comp $ f.to_add_monoid_hom.continuous_completion f.continuous)
+            (continuous_const.mul continuous_norm) },
+    { intro x,
+      change ∥f.to_add_monoid_hom.completion _ ↑x∥ ≤ ∥f∥ * ∥↑x∥,
+      rw f.to_add_monoid_hom.completion_coe f.continuous,
+      simp only [completion.norm_coe],
+      exact f.le_op_norm x }
+  end,
+  ..f.to_add_monoid_hom.completion f.continuous }
+
+lemma normed_group_hom.completion_def (f : normed_group_hom G H) (x : completion G) :
+  f.completion x = completion.map f x := rfl
+
+lemma normed_group_hom.completion_coe_to_fun (f : normed_group_hom G H) :
+  (f.completion : completion G → completion H) = completion.map f :=
+by { ext x, exact normed_group_hom.completion_def f x }
+
+@[simp]
+lemma normed_group_hom.completion_coe (f : normed_group_hom G H) (g : G) : f.completion g = f g :=
+completion.map_coe f.uniform_continuous _
+
+/-- Completion of normed group homs as a normed group hom. -/
+def normed_group_hom_completion_hom :
+  normed_group_hom G H →+ normed_group_hom (completion G) (completion H) :=
+{ to_fun := normed_group_hom.completion,
+  map_zero' := begin
+    apply to_add_monoid_hom_injective,
+    exact add_monoid_hom.completion_zero
+  end,
+  map_add' := λ f g, begin
+    apply to_add_monoid_hom_injective,
+    exact f.to_add_monoid_hom.completion_add g.to_add_monoid_hom f.continuous g.continuous,
+  end }
+
+@[simp]
+lemma normed_group_hom.completion_id :
+  (normed_group_hom.id G).completion = normed_group_hom.id (completion G) :=
+begin
+  ext x,
+  rw [normed_group_hom.completion_def, normed_group_hom.coe_id, completion.map_id],
+  refl
+end
+
+lemma normed_group_hom.completion_comp (f : normed_group_hom G H) (g : normed_group_hom H K) :
+  g.completion.comp f.completion = (g.comp f).completion :=
+begin
+  ext x,
+  rw [normed_group_hom.coe_comp, normed_group_hom.completion_def,
+    normed_group_hom.completion_coe_to_fun, normed_group_hom.completion_coe_to_fun,
+    completion.map_comp (normed_group_hom.uniform_continuous _)
+    (normed_group_hom.uniform_continuous _)],
+  refl
+end
+
+lemma normed_group_hom.completion_neg (f : normed_group_hom G H) :
+  (-f).completion = -f.completion :=
+normed_group_hom_completion_hom.map_neg f
+
+lemma normed_group_hom.completion_add (f g : normed_group_hom G H) :
+  (f + g).completion = f.completion + g.completion :=
+normed_group_hom_completion_hom.map_add f g
+
+lemma normed_group_hom.completion_sub (f g : normed_group_hom G H) :
+  (f - g).completion = f.completion - g.completion :=
+normed_group_hom_completion_hom.map_sub f g
```

Comment:
Should these be simp-lemmas?

Reply:
I don't think so. It's really not clear which side is simpler.